text/0092-logs-vision.md

```diff
@@ -73,6 +73,15 @@ Unified collection is important for the following reasons:
   of resources where the data comes from or by user-defined attributes),
   enabling correct correlation across Resource dimensions later on the backend.
 
+#### Performance
+
+The unified collector must be performant; its resource utilization should be minimal for most use cases, but must also handle high throughput when needed.
+
+We have identified three sets of use cases for the collector with differing requirements:
+- *Side-car*: There are a set of use cases where a single instance of the collector will be deployed alongside a single instance of an application- a 1:1 relationship. This set includes many serverless environments. A single instance of an application may produce telemetry events (logs, metrics, traces) measured in 10s to 1000s per second. For these use cases, the collector's resource consumption should not exceed 100 MB memory, or 20% of a CPU on a modern server. We will try to optimize the collector so that for many users in this group, the collector will consume less than 50 MB of memory and less than 10% of a CPU.
+- *Daemon*: Many users will deploy the collector as a daemon, a single instance per node or host, to collect telemetry from multiple applications. In containerized environments, [roughly 10-20 containers are generally run per node](https://www.datadoghq.com/docker-adoption/). As noted above, these applications may emit 10s to 1000s of telemetry events per second. For this segment, our goal is that the collector not be a significant consumer of resources when compared with applications deployed on that node. Its resource consumption should be a fraction of the usage of the applications that it serves; preferably <10%, ideally <5%.
+- *Service*: The collector may also be deployed as a standalone service that serves multiple applications across multiple nodes. For this use case, low resource consumption is not the primary concern; high throughput is the goal.
```

Comment:
>low resource consumption is not the primary concern

I know people for whom resource consumption in standalone mode is a significant concern, to the point that they are willing to ditch the Collector and write a custom service to proxy telemetry data faster than the Collector can do today.

I would delete the first part of the statement and perhaps only mention the "throughout" part, which is a valid concern. Additionally, throughput needs to be quantified. It is highly dependent on the protocol and on roundtrip latency.